python/tests/test_logistic_regression.py

```diff
@@ -1986,3 +2021,84 @@ def sparse_to_df(X: csr_matrix, y: List[float]) -> DataFrame:
             tolerance,
             accuracy_and_probability_only=accuracy_and_probability_only,
         )
+
+
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize(
+    "reg_factors", [(0.0, 0.0), (0.1, 0.0), (0.1, 1.0), (0.1, 0.2)]
+)
+@pytest.mark.parametrize("feature_type", ["vector"])
+@pytest.mark.parametrize("data_shape", [(100, 8)], ids=idfn)
+@pytest.mark.parametrize("max_record_batch", [20])
+@pytest.mark.parametrize("n_classes", [2, 4])
+@pytest.mark.slow
+def test_double_precision(
+    fit_intercept: bool,
+    reg_factors: Tuple[float, float],
+    feature_type: str,
+    data_shape: Tuple[int, int],
+    max_record_batch: int,
+    n_classes: int,
+    gpu_number: int,
+) -> None:
+
+    random_bool = random.choice([True, False])
```

Comment:
Would be good to make these deterministic via seed for reproducibility.

Reply:
Done.